bx_py_utils/hash_utils.py

```diff
@@ -0,0 +1,55 @@
+import hashlib
+import itertools
+import string
+
+
+class UrlSafeEncoder:
+    """
+    Encode bytes into a URL safe string.
+    Note:
+        Use a URL safe alphabet (see RFC 3986) without umlauts
+
+    Remove umlauts will hopefully avoid to generate valid words ;)
+    """
+    alphabet = (
+        '-._~'  # Allowed additional characters RFC 3986
+        'bcdfghjklmnpqrstvwxyz'  # lower case letters without aeiou
+        'BCDFGHJKLMNPQRSTVWXYZ'  # upper case letters without AEIOU
+    ) + string.digits
+    _map = None
+
+    @classmethod
+    def encode(cls, data):
+        assert isinstance(data, bytes)
+
+        if cls._map is None:
+            # Delay the initialization of the map to not waste memory if encoder never used
+            cls._map = {i: char for i, char in zip(range(256), itertools.cycle(cls.alphabet))}
+
+        return ''.join(cls._map[char] for char in data)
```

Comment:
If `alphabet` is a list or tuple, then `alphabet[char % len(alphabet)]` is much more readable.

Reply:
Hm! That also gives me an idea to refactor the class and make a simple function...

btw. i timit the index lookup between string, tuple and dict: string is the fastest (and the simples implementation)